### PR TITLE
Rowan edits

### PR DESCRIPTION
**Pull request by rowandeer**
Here are the agreed updates to 4 and 5, and my edits to 6.

Here are the notes to 6:

“Hint 3: The code that's already there can stay as it is. However, it needs to be **inside** a `while` loop. So you need to add one line above all the other code to create a `while` loop. And because the existing code needs to be inside the `while` loop's **code block** you need to indent **all** the following lines using a tab or two spaces.”

This is confusing to me because it’s not how I solved it. I solved it like this:

```
user_password = ""

# 👇 Make your edits below this line: 


user_password = input("Please choose a password:")

while len(user_password) < 3:
    print("The password is too short. Please choose a longer password.")
    user_password = input("Please choose a password:")

print("Password set!")

user_password = ""
```

i.e. my while loop does not start above all the other code (and it does seem to work)… is this wrong? Or does the hint need to be corrected?

—

“In case you were wondering: We defined `user_password` in the example above as an empty string first. That's because we use that variable within the loop condition. You can only use variables if they have been defined already. Without that, the code would throw an error. You should try it out! Remove the first line and run the code from above again. Read the error message and try to make sense of it. You'll likely encounter this error in the future again.”

Again, with the way I solved it, this is not true (because I’ve defined the variable as user input outside the while loop…)

—

“You also need to pay attention to when each variable is defined. Look at this example code: 

```python
while favorite_food != "pizza"
  favorite_food = input("What is your favorite food?")
```

This code will break. Why? Because you're using the `favorite_food` variable before it has been declared. You can solve that issue by defining the variable as something else (e.g., an empty string) at the start of the program, like so: 

```python
favorite_food = ""

while favorite_food != "pizza"
  favorite_food = input("What is your favorite food?”)”
```

Again, my instinct was to declare the variable with an input outside of the loop (pasted below), so again this hint seems slightly off. Wondering if I’m gone wrong (and if so do we need to add guidance so users don’t do the same as I did) or whether we need to accommodate this possibility in the hint? (My code works)

```
password = "supersecret"

# TODO: Add your code below

user_input = input("Please enter the password:")
while user_input != password:
    user_input = input("Sorry, that's not right, please try again:")

print("You are logged in.")
```

—

“**What do you think will happen?**” quiz

Is the use supposed to be able to select one of these? That was my instinct and then when I try to click on it it turns into a markdown code, which I guess is not correct. Wondering why this isn’t just in the text cell above or what it’s supposed to do.

—

“Ensure all the code is inside a `while True:` loop (including the last print and the `user_age` declaration).”

Added the “while True:” bit because my instinct was to just try to use while instead of if - I think because “while True:” got introduced as bad practice I wasn’t thinking about the possibility of using it  also deleted the parenthesis here, because as far as I can tell the last print statement should not be within the loop? i.e. the following works, but when I put the last print statement inside the loop, then it doesn’t get printed :

—

"**Quiz Game (Advanced)**

Similarly, you can improve on the guessing game from the previous exercises. In the previous lessons, you built a game where users could guess the height of Mount Everest. Try building a proper quiz game with up to 5 questions. Make use of loops and functions so you don't end up with lots of redundant code. 

This task is a bit more open-ended and more difficult to complete. But you can use it as a challenge for yourself."

[deleted this bit since we added it to the previous lesson]

—
"**Rock-Paper-Scissors Improvement**"
 [I got the basic game to work in the past lesson but I’m definitely struggling with this. I think some more guidance is needed?]

### CODE REVIEW

**Comment by samlawski** [comment]
**About the first password lesson**: Interesting solution. You basically solved it a bit more complicated than it needed to be. All you need to do is insert a single line of code and indent the rest: 

```python
user_password = ""

# 👇 Make your edits below this line: 

while len(user_password) < 3:
  user_password = input("Please choose a password:")

  if len(user_password) < 3:
    print("The password is too short. Please choose a longer password.")
  else:
    print("Password set!")
```

This solution has less repetition and is a bit simpler. But yours works, too.

**About the second password exercise**: There will be regular situations where you solve something differently from what I had thought of originally. I think that's perfectly fine and normal. Maybe we should then rephrase it to say: "One approach is to ..."

**About the food hint:** Same as above. You just thought of solving it differently. I tried to simplify by using `input()` only once. But you also used different prompt text. So your solution is perfectly fine, too. Defining an "empty variable" is a common pattern used occasionally by developers. So I believe it's still useful to see as an example. 

**About “What do you think will happen?” quiz**: Ah good point. I only edited it in VSCode and didn't notice that it would enter edit mode when you click it. That's obviously not ideal. We can also remove the interactivity and just make it a bullet list because it doesn't actually do anything if you select an option. I just thought it might be a fun little interactive element. But if it causes a risk of confusing people let's rather take it out. 

**About get_user_number function**: Again, it's perfectly fine to come to a different solution. Usually there are quite a lot of different ways to solve the same task. In this particular case, the learner was even encouraged not to follow the instructions. So I feel like it's ok that you found a solution that deviates from the instructions. 

**About rock-paper-scissors**: Alright I'll add a task to review this exercise to possible add more detailed instructions.

**Comment by rowandeer** [comment]
Oops, I totally should have read through my notes more thoroughly before hitting send... with the get_user_number bit I actually went back and realised the point I was tryna make didn't stand. have edited my comment to delete that bit.

RE the other comments, do we just need to add something to an earlier lesson &/ our introduction that reiterates that the hints will often guide you to one solution, but if you've already got to a solution in a different way that works, then that's fine?

**Comment by samlawski** [comment]
Yes I think that's an option. Do you know a good spot to add it and would you like to do so? Otherwise I can have another look later.
